internal/provider/models/alert_route_v1_model.go

```diff
@@ -0,0 +1,575 @@
+				model.SlackTargets = &AlertRouteChannelTargetModel{
+					ChannelVisibility: types.StringValue(channelConfig.SlackTargets.ChannelVisibility),
+					Binding:           &binding,
+				}
+			}
+
+			// Convert MS Teams targets
+			if channelConfig.MsTeamsTargets != nil {
+				binding := IncidentEngineParamBinding{}.FromAPI(channelConfig.MsTeamsTargets.Binding)
+				model.MsTeamsTargets = &AlertRouteChannelTargetModel{
+					ChannelVisibility: types.StringValue(channelConfig.MsTeamsTargets.ChannelVisibility),
+					Binding:           &binding,
+				}
+			}
+
+			result.ChannelConfig = append(result.ChannelConfig, model)
+		}
+	} else {
+		// Ensure we return an empty array instead of null
+		result.ChannelConfig = []AlertRouteChannelConfigModel{}
+	}
+
+	// Convert top-level condition groups - use engine model
+	if apiModel.ConditionGroups != nil {
+		result.ConditionGroups = IncidentEngineConditionGroups{}.FromAPI(*apiModel.ConditionGroups)
+	} else {
+		// Ensure we return an empty array instead of null
+		result.ConditionGroups = IncidentEngineConditionGroups{}
+	}
+
+	// Convert expressions - use engine model
+	// ALWAYS set expressions to an initialized empty slice
+	result.Expressions = IncidentEngineExpressions{}
+	if apiModel.Expressions != nil && len(*apiModel.Expressions) > 0 {
+		result.Expressions = IncidentEngineExpressions{}.FromAPI(*apiModel.Expressions)
+	}
+
+	// Convert escalation config
+	if apiModel.EscalationConfig != nil {
+		result.EscalationConfig = &AlertRouteEscalationConfigModel{
+			AutoCancelEscalations: types.BoolValue(apiModel.EscalationConfig.AutoCancelEscalations),
+			EscalationTargets:     []AlertRouteEscalationTargetModel{},
+		}
+
+		// Convert escalation targets
+		if apiModel.EscalationConfig.EscalationTargets != nil {
+			for _, target := range apiModel.EscalationConfig.EscalationTargets {
+				model := AlertRouteEscalationTargetModel{}
+
+				if target.Users != nil {
+					binding := IncidentEngineParamBinding{}.FromAPI(*target.Users)
+					model.Users = &binding
+				}
+
+				if target.EscalationPaths != nil {
+					binding := IncidentEngineParamBinding{}.FromAPI(*target.EscalationPaths)
+					model.EscalationPaths = &binding
+				}
+
+				result.EscalationConfig.EscalationTargets = append(result.EscalationConfig.EscalationTargets, model)
+			}
+		}
+	}
+
+	// Convert incident config
+	if apiModel.IncidentConfig != nil {
+		result.IncidentConfig = &AlertRouteIncidentConfigModel{
+			AutoDeclineEnabled: types.BoolValue(apiModel.IncidentConfig.AutoDeclineEnabled),
+			DeferTimeSeconds:   types.Int64Value(apiModel.IncidentConfig.DeferTimeSeconds),
+			Enabled:            types.BoolValue(apiModel.IncidentConfig.Enabled),
+			GroupingKeys:       []AlertRouteGroupingKey{},
+		}
+
+		// Convert grouping keys
+		for _, gk := range apiModel.IncidentConfig.GroupingKeys {
+			groupingKey := AlertRouteGroupingKey{
+				ID: types.StringValue(gk.Id),
+			}
+
+			if gk.Reference != nil {
```

Comment:
yh this feels weird to me: it's marked required in the schema but we aren't always setting it 🤔